src/main/java/org/dependencytrack/notification/NotificationRouter.java

```diff
@@ -219,4 +219,20 @@ private void limitToProject(final List<NotificationRule> applicableRules, final
         }
     }
 
+    private boolean checkIfChildrenAreAffected(Project parent, UUID uuid){
+        boolean isChild = false;
+        if (parent.getChildren() == null || parent.getChildren().isEmpty()){
+            return false;
+        } else {
+            for (Project child : parent.getChildren()){
+                if ((child.getUuid().equals(uuid) && child.isActive()) || isChild){
+                    return true;
+                } else {
+                    isChild = checkIfChildrenAreAffected(child, uuid);
+                }
+            }
+        }
```

Comment:
```suggestion
        }
        for (Project child : parent.getChildren()){
            if ((child.getUuid().equals(uuid) && child.isActive()) || isChild){
                return true;
            }
            isChild = checkIfChildrenAreAffected(child, uuid);
        }
```
avoid else after return to reduce nesting